include/circt/Dialect/FIRRTL/FIRRTLAnnotations.h

```diff
@@ -123,8 +123,8 @@ class AnnotationSet {
   /// Get an annotation set for the specified operation.
   explicit AnnotationSet(Operation *op);
 
-  /// Get an annotation set for the specified module port.
-  static AnnotationSet forPort(FModuleLike module, size_t portNo);
+  /// Get an annotation set for the specified port.
+  static AnnotationSet forPort(Operation *op, size_t portNo);
```

Comment:
It might be worth being conservative here and having a FModuleLike and a MemOp constructor since this attribute is only really defined on those ops.

Reply:
I split the API:
```c++
  /// Get an annotation set for the specified port.
  static AnnotationSet forPort(FModuleLike op, size_t portNo);
  static AnnotationSet forPort(MemOp op, size_t portNo);
```